lib/tramway/base_form.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+module Tramway
+  # Provides form object for Tramway
+  #
+  class BaseForm
+    attr_reader :object
+
+    %i[model_name to_key to_model errors attributes].each do |method_name|
+      delegate method_name, to: :object
+    end
+
+    def initialize(object)
+      @object = object
+    end
+
+    class << self
+      def property(attribute, _proc_obj = nil)
+        @properties ||= []
+        @properties << attribute
+
+        delegate attribute, to: :object
+      end
+
+      def properties(*attributes)
+        if attributes.any?
+          attributes.each do |attribute|
+            property(attribute)
+          end
+        else
+          @properties || []
+        end
+      end
+    end
+
+    def submit(params)
+      __submit params
+
+      object.save.tap do
+        object.reload
+      end
+    end
+
+    def submit!(params)
+      __submit params
+
+      object.save!.tap do
+        object.reload
+      end
+    end
+
+    def method_missing(method_name, *args)
+      if method_name.to_s.end_with?('=') && args.count == 1
+        object.public_send(method_name, args.first)
+      else
+        super
+      end
+    end
```

Comment:
`respond_to_missing?` should also be defined

Reply:
Fixed!

Thanks!